shell/platform/android/io/flutter/view/VsyncWaiter.java

```diff
@@ -14,11 +16,27 @@
     // The initial value of 0.0 indicates unkonwn refresh rate.
     public static float refreshRateFPS = 0.0f;
 
+    private static HandlerThread handlerThread;
+    private static Handler handler;
+
+    static {
+        handlerThread = new HandlerThread("FlutterVsyncThread");
+        handlerThread.start();
+    }
+
     public static void asyncWaitForVsync(final long cookie) {
-        Choreographer.getInstance().postFrameCallback(new Choreographer.FrameCallback() {
+        if (handler == null) {
+            handler = new Handler(handlerThread.getLooper());
```

Comment:
Can the handler be assigned from the looper for the current thread? Instead of creating another thread I mean. This thread has a native looper.

Reply:
For reference, the native looper is created here. https://github.com/flutter/engine/blob/930033d95c8f43e17020d651db5f5997fc8fcb6c/fml/platform/android/message_loop_android.cc#L22

Reply:
Do you mean to use AChoreographer? But its min api level is 24.
```
#include <sys/cdefs.h>

__BEGIN_DECLS

#if __ANDROID_API__ >= 24

struct AChoreographer;
typedef struct AChoreographer AChoreographer;

/**
 * Prototype of the function that is called when a new frame is being rendered.
 * It's passed the time that the frame is being rendered as nanoseconds in the
 * CLOCK_MONOTONIC time base, as well as the data pointer provided by the
 * application that registered a callback. All callbacks that run as part of
 * rendering a frame will observe the same frame time, so it should be used
 * whenever events need to be synchronized (e.g. animations).
 */
typedef void (*AChoreographer_frameCallback)(long frameTimeNanos, void* data);
```

Reply:
I can use AChoreographer when API Level >= 24, but lower API Level still use a new handler thread.